BoundedKnapsack/ReferenceImplementation.qs

```diff
@@ -0,0 +1,406 @@
+                Controlled X(a[i..D-1], target);
+                // Flips the qubit if b's corresponding bit is 0.
+                // This temporarily sets the qubit to 1 if the corresponding bits are equal.
+                X(a[i]);
+            }
+        }
+
+        // Uncompute
+        ApplyPauliFromBitString(PauliX, false, binaryB, a);
+    }
+
+
+    // Task 1.4. Compare Qubit Array with Integer (≤)
+    operation CompareQubitArrayLeqThanInt_Reference (a : Qubit[], b : Int, target : Qubit) : Unit is Adj+Ctl {
+        // This operation essentially calculates the opposite of the greater-than
+        // comparator, so we can just call CompareQubitArrayGreaterThanInt, and then an X gate.
+        CompareQubitArrayGreaterThanInt_Reference(a, b, target);
+        X(target);
+    }
+
+
+    // Task 1.5. Verify that total weight doesn't exceed limit W
+    operation VerifyWeight_01_Reference (n: Int, W : Int, maxTotal : Int, itemWeights : Int[], register : Qubit[], target : Qubit) : Unit is Adj+Ctl {
+        using (totalWeight = Qubit[maxTotal]) {
+            within {
+                CalculateTotalWeightOrProfit_01_Reference(n, itemWeights, register, totalWeight);
+            } apply {
+                CompareQubitArrayLeqThanInt_Reference(totalWeight, W, target);
+            }
+        }
+    }
+
+
+    // Task 1.6. Verify that the total profit exceeds threshold P
+    operation VerifyProfit_01_Reference (n: Int, P : Int, maxTotal : Int, itemProfits : Int[], register : Qubit[], target : Qubit) : Unit is Adj+Ctl {
+        using (totalProfit = Qubit[maxTotal]) {
+            within {
+                CalculateTotalWeightOrProfit_01_Reference(n, itemProfits, register, totalProfit);
+            } apply {
+                CompareQubitArrayGreaterThanInt_Reference(totalProfit, P, target);
+            }
+        }
+    }
+
+
+    // Task 1.7. 0-1 Knapsack Problem Validation Oracle
+    operation KnapsackValidationOracle_01_Reference (n : Int, W : Int, P : Int, maxTotal : Int, itemWeights : Int[], itemProfits : Int[], register : Qubit[], target : Qubit) : Unit is Adj+Ctl {
+        using ((outputW, outputP) = (Qubit(), Qubit())) {
+            within {
+                VerifyWeight_01_Reference(n, W, maxTotal, itemWeights, register, outputW);
+                VerifyProfit_01_Reference(n, P, maxTotal, itemProfits, register, outputP);
+            } apply {
+                CCNOT(outputW, outputP, target);
+            }
+        }
+    }
+
+
+    //////////////////////////////////////////////////////////////////
+    // Part II. Bounded Knapsack Problem
+    //////////////////////////////////////////////////////////////////
+    
+
+    // Task 2.1. Read combination from a jagged array of qubits
+    operation MeasureCombination_Reference (xs : Qubit[][]) : Int[] {
+        return Mapped(Compose(ResultArrayAsInt, MultiM), xs);

```

Comment:
@emax-wenjun Sorry, this was my mistake. I suggest you revert to the previous version for this one. I didn't take into account that `MultiM` is an operation and not a function, but `Compose` requires two functions as argument.